### PR TITLE
use static where possible

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

File: GitCommands/CustomDiffMergeToolCache.cs [hunk 1]
```diff
@@ -147,7 +147,7 @@ public TestAccessor(CustomDiffMergeToolCache cache)
             }
 
             public IEnumerable<string> ParseCustomDiffMergeTool(string output, string defaultTool)
-                => _cache.ParseCustomDiffMergeTool(output, defaultTool);
+                => CustomDiffMergeToolCache.ParseCustomDiffMergeTool(output, defaultTool);
```
Comment (gerhardol) [inline GitCommands/CustomDiffMergeToolCache.cs:150]: Remove _cache too

File: GitCommands/DiffMergeTools/DiffMergeToolConfigurationManager.cs [hunk 2]
```diff
@@ -235,7 +235,7 @@ public TestAccessor(DiffMergeToolConfigurationManager manager)
             }
 
             public (string toolKey, string prefix) GetInfo(DiffMergeToolType toolType)
-                => _manager.GetInfo(toolType);
+                => DiffMergeToolConfigurationManager.GetInfo(toolType);
```
Comment (gerhardol) [inline GitCommands/DiffMergeTools/DiffMergeToolConfigurationManager.cs:238]: _manager unused

File: GitUI/CommandsDialogs/FormClone.cs [hunk 3]
```diff
@@ -493,7 +493,7 @@ public TestAccessor(FormClone form)
                 _form = form;
             }
 
-            public bool TryExtractUrl(string text, out string url) => _form.TryExtractUrl(text, out url);
+            public bool TryExtractUrl(string text, out string url) => FormClone.TryExtractUrl(text, out url);
```
Comment (gerhardol) [inline GitUI/CommandsDialogs/FormClone.cs:496]: _form unused

File: ResourceManager/LinkFactory.cs [hunk 4]
```diff
@@ -174,10 +174,10 @@ public TestAccessor(LinkFactory linkFactory)
             }
 
             public bool ParseInternalScheme(Uri uri, [NotNullWhen(returnValue: true)] out CommandEventArgs? commandEventArgs)
-                => _linkFactory.ParseInternalScheme(uri, out commandEventArgs);
+                => LinkFactory.ParseInternalScheme(uri, out commandEventArgs);
 
             public bool TryParseLink(string? linkUri, [NotNullWhen(returnValue: true)] out Uri? uri)
-                => _linkFactory.TryParseLink(linkUri, out uri);
+                => LinkFactory.TryParseLink(linkUri, out uri);
```
Comment (gerhardol) [inline ResourceManager/LinkFactory.cs:180]: _linkFactory unused

**Review by gerhardol — COMMENTED** [review]
Brief review
@RussKie normally request that functions are not static though

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
> @RussKie normally request that functions are not static though

I am very much against public static types and API because those are PIA to test at callsites. These changes appears to only modify private API, so 👍 
Thanks

**Review by mstv — COMMENTED** [review]
👍
A merge conflict needs to be resolved

**Comment by gerhardol** [comment]
@SimonCropp Can you resolve the merge conflicts or do you want a maintainer to fix it?
(Trivial fix, the changed method was removed.)

**Comment by SimonCropp** [comment]
conflict resolved

**Review by mstv — APPROVED** [review]

**Pull request merged**